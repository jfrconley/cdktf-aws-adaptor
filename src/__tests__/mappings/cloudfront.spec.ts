import { CloudfrontOriginAccessIdentity } from "@cdktf/provider-aws/lib/cloudfront-origin-access-identity/index.js";
import { CfnCloudFrontOriginAccessIdentity } from "aws-cdk-lib/aws-cloudfront";
import { itShouldMapCfnElementToTerraformResource } from "../helpers.js";

describe("CloudFront", () => {
  itShouldMapCfnElementToTerraformResource(
    CfnCloudFrontOriginAccessIdentity,
    {
      cloudFrontOriginAccessIdentityConfig: {
        comment: "test",
      },
    },
    CloudfrontOriginAccessIdentity,
    {
      comment: "test",
    },
  );
});
