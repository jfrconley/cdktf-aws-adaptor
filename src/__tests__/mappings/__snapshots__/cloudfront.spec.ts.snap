// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CloudFront > Should map AWS::CloudFront::CloudFrontOriginAccessIdentity 1`] = `
"{
  "provider": {
    "aws": [
      {
        "alias": "us_east_1",
        "region": "us-east-1"
      }
    ]
  },
  "resource": {
    "aws_cloudfront_origin_access_identity": {
      "resource_22C949BF": {
        "comment": "test"
      }
    }
  },
  "terraform": {
    "backend": {
      "local": {
        "path": "/terraform.test-stack.tfstate"
      }
    },
    "required_providers": {
      "aws": {
        "source": "aws",
        "version": "5.85.0"
      }
    }
  },
  "variable": {
    "resource_refs_CDA17697": {
      "default": "\${join(\\"\\", [aws_cloudfront_origin_access_identity.resource_22C949BF.id, \\",\\", aws_cloudfront_origin_access_identity.resource_22C949BF.s3_canonical_user_id])}",
      "type": "string"
    }
  }
}"
`;
